addons/debugger/userscript.js

```diff
@@ -166,336 +368,235 @@ export default async function ({ addon, global, console, msg }) {
+    let shape;
+    let color;
+    if (
+      block.opcode === "data_variable" ||
+      block.opcode === "data_listcontents" ||
+      block.opcode === "argument_reporter_string_number" ||
+      block.opcode === "argument_reporter_boolean"
+    ) {
+      text = Object.values(block.fields)[0].value;
+      if (block.opcode === "data_variable") {
+        category = "data";
+      } else if (block.opcode === "data_listcontents") {
+        category = "list";
+      } else {
+        category = "more";
+      }
+      shape = "round";
+    } else if (block.opcode === "procedures_call") {
+      const proccode = block.mutation.proccode;
+      text = formatProcedureCode(proccode);
+      const customBlock = addon.tab.getCustomBlock(proccode);
+      if (customBlock) {
+        category = "addon-custom-block";
+        color = customBlock.color;
+      } else {
+        category = "more";
+      }
+    } else if (block.opcode === "procedures_definition") {
+      const prototypeBlockId = block.inputs.custom_block.block;
+      const prototypeBlock = target.blocks.getBlock(prototypeBlockId);
+      const proccode = prototypeBlock.mutation.proccode;
+      text = ScratchBlocks.ScratchMsgs.translate("PROCEDURES_DEFINITION", "define %1").replace(
+        "%1",
+        formatProcedureCode(proccode)
+      );
+      category = "more";
+    } else {
+      // Try to call things like https://github.com/LLK/scratch-blocks/blob/0bd1a17e66a779ec5d11f4a00c43784e3ac7a7b8/blocks_vertical/operators.js#L36
+      var jsonData;
+      const fakeBlock = {
+        jsonInit(data) {
+          jsonData = data;
+        },
+      };
+      const blockConstructor = ScratchBlocks.Blocks[block.opcode];
+      if (blockConstructor) {
+        try {
+          blockConstructor.init.call(fakeBlock);
+        } catch (e) {
+          // ignore
+        }
+      }
+      if (!jsonData) {
+        return null;
+      }
+      text = formatBlocklyBlockData(jsonData);
+      if (!text) {
+        return null;
+      }
+      category = jsonData.category;
+      const isStatement =
+        (jsonData.extensions &&
+          (jsonData.extensions.includes("shape_statement") ||
+            jsonData.extensions.includes("shape_hat") ||
+            jsonData.extensions.includes("shape_end"))) ||
+        "previousStatement" in jsonData ||
+        "nextStatement" in jsonData;
+      shape = isStatement ? "stacked" : "round";
+    }
+    if (!text || !category) {
+      return null;
+    }
 
-  function elementDrag(e) {
-    e.preventDefault();
-    dragConsole(e.clientX, e.clientY);
-  }
+    if (!color) {
+      const blocklyCategoryMap = {
+        "data-lists": "data_lists",
+        list: "data_lists",
+        events: "event",
+      };
+      const blocklyColor = ScratchBlocks.Colours[blocklyCategoryMap[category] || category];
+      if (blocklyColor) {
+        // block probably belongs to an extension
+        color = blocklyColor.primary;
+      } else {
```

Comment:
I forgot to move this comment after reorganizing

```suggestion
        color = blocklyColor.primary;
      } else {
        // block probably belongs to an extension
```